content/webapp/pages/search/index.tsx

```diff
@@ -305,59 +318,171 @@ const StoryPromoContainer = styled(Container)`
 `;
 
 const NewSearchPage: NextPageWithLayout<NewProps> = ({
+  query,
   queryString,
   contentResults,
-  catalogueResults,
   contentQueryFailed,
 }) => {
   const { extraApiToolbarLinks, setExtraApiToolbarLinks } =
     useContext(SearchContext);
   const { apiToolbar } = useToggles();
+  const params = fromQuery(query);
+  const data = useContext(ServerDataContext);
+  const { setLink } = useContext(SearchContext);
+  const [clientSideWorkTypes, setClientSideWorkTypes] = useState<
+    WorkTypes | undefined
+  >(undefined);
+  const [clientSideImages, setClientSideImages] = useState<
+    ImageResults | undefined
+  >(undefined);
 
-  const totalWorksResults = catalogueResults.works?.totalResults || 0;
-  const totalImagesResults = catalogueResults.images?.totalResults || 0;
+  const [isLoadingImages, setIsLoadingImages] = useState(true);
+  const [isLoadingWorks, setIsLoadingWorks] = useState(true);
+  const [isCatalogueLoading, setIsCatalogueLoading] = useState(true);
+  const [contentResultsLength, setContentResultsLength] = useState(
+    contentResults?.totalResults
+  );
+
+  const totalWorksResults = clientSideWorkTypes?.totalResults || 0;
+  const totalImagesResults = clientSideImages?.totalResults || 0;
   const totalContentResults = contentResults?.totalResults || 0;
   const totalCatalogueResults = totalWorksResults + totalImagesResults;
   const totalResults = totalContentResults + totalCatalogueResults;
   const isSmallGallery =
-    catalogueResults.images && catalogueResults.images.results.length < 3;
+    clientSideImages && clientSideImages.results.length < 3;
   const pathname = usePathname();
 
+  useMemo(() => {
+    if (contentResultsLength !== contentResults?.totalResults)
+      setIsCatalogueLoading(true);
+    setContentResultsLength(contentResults?.totalResults);
+  }, [contentResults?.totalResults]);
```

Comment:
I had to do this because the content results would load faster (of course) than the loading state could go back to true. If you comment this bit, you'll see that previous results stay of a bit before displaying the loading state, which looks especially strange when moving from a grid to a full-size.